template/clubhouse.py

```diff
@@ -88,3 +93,34 @@ def _proxy_ready(proxy):
             proxy.connect('g-properties-changed', _props_changed_cb, *args)
 
         klass.get_proxy_async(_proxy_ready)
+
+
+class GnomeSoftware:
+    _proxy = None
+
+    @classmethod
+    def get_proxy(klass):
+        if klass._proxy is None:
+            klass._proxy = \
+                Gio.DBusProxy.new_for_bus_sync(Gio.BusType.SESSION,
+                                               Gio.DBusProxyFlags.DO_NOT_AUTO_START |
+                                               Gio.DBusProxyFlags.DO_NOT_AUTO_START_AT_CONSTRUCTION,
+                                               None,
+                                               'org.gnome.Software',
+                                               '/org/gnome/Software',
+                                               'org.gtk.Actions',
+                                               None)
+        return klass._proxy
+
+    @classmethod
+    def details(klass, app_id):
+        def _on_done_callback(proxy, result):
+            try:
+                proxy.call_finish(result)
+            except GLib.Error as e:
+                logger.error('Error showing details of app: %s', e.message)
+
+        args = GLib.Variant('(ss)', (app_id, ''))
+        variant = GLib.Variant('(sava{sv})', ['details', [args], None])
+        klass.get_proxy().call('Activate', variant,
```

Comment:
I think that you can get rid of the variant conversion if you use the higher-level `Gio.DBusProxy.makeProxyWrapper` to obtain the proxy, and then call `proxy.ActivateRemote(...)`.